newlib/libc/sys/vita/syscalls.c

```diff
@@ -324,6 +333,14 @@ _read_r(struct _reent *reent, int fd, void *ptr, size_t len)
 		case VITA_DESCRIPTOR_DIRECTORY:
 			ret = __vita_make_sce_errno(EBADF);
 			break;
+		case VITA_DESCRIPTOR_PIPE:
+		{
+			size_t rlen = len;
+			if (rlen > 4096) rlen = 4096;
+			ret = sceKernelReceiveMsgPipe(fdmap->sce_uid, ptr, rlen, 1, NULL, NULL);
```

Comment:
if `sceKernel(Send/Receive)MsgPipe` require set 0x1000 for buf size & we support large pipe buffer, it needs repeating

Reply:
No. Read/write returns number of actual written/read bytes and doesn't guarantee (per standard) that all bytes can be read/written (specifically in case of pipe). It's up to application to repeat

Reply:
i mean it's related part of the `open_r`
if we support large pipe buffer aka `big pipe`, system could allow read/write large buffer at one time even if `PIPE_BUF` is 4k.
btw, i rechecked `PIPE_BUF` part in the [write function](https://pubs.opengroup.org/onlinepubs/9699919799/functions/write.html), you are right. it doesn't guarantee in spec.